### PR TITLE
add update-docs workflow

### PR DESCRIPTION
**Pull request by aaperis**
This is needed for migrating the automatic documentation update from `sda-pipeline` to this repository.

### CODE REVIEW

**Review by blankdots — APPROVED** [review]

**Review by pontus — DISMISSED** [review]

**Review by dbampalikis — DISMISSED** [review]

File: .github/workflows/update-docs.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+    name: Trigger documentation rebuild in neic-sda
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          result=$(curl -X POST \
+          -H "Authorization: Bearer ${{secrets.NEIC_SDA_ACCESS}}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          https://api.github.com/repos/neicnordic/neic-sda/actions/workflows/aggregate.yml/dispatches \
+          -d "{\"ref\": \"master\", \"inputs\":{\"repository\": \"$GITHUB_REPOSITORY\"}}")
```
Comment (kusalananda) [inline .github/workflows/update-docs.yml:24]: Indent for readability, put URL last, and sort out quoting of the JSON payload.

```suggestion
          result=$(
              curl -X POST \
                 -H "Authorization: Bearer ${{secrets.NEIC_SDA_ACCESS}}" \
                 -H "Accept: application/vnd.github.v3+json" \
                 -d '{"ref": "master", "inputs": {"repository": "'"$GITHUB_REPOSITORY"'"}}' \
                 https://api.github.com/repos/neicnordic/neic-sda/actions/workflows/aggregate.yml/dispatches
          )

```

File: .github/workflows/update-docs.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          [ -z "$result" ] && true || (echo "$result" && false)
```
Comment (kusalananda) [inline .github/workflows/update-docs.yml:25]: `&& true` is a no-op. Write it with a `if` so it's clear what we want to happen.

```suggestion
          if [ -n "$result" ]; then
              echo "$result"
              false
          fi
```

**Review by kusalananda — COMMENTED** [review]
Nitpicking, for readability. Ignore it if it's not worth the effort.

File: .github/workflows/update-docs.yml (hunk 1 above)
Comment (aaperis) [reply .github/workflows/update-docs.yml:24]: To be frank, I don't find `'"'"$a"'"'` more readable as compared to `"\"$a\""` so I would prefer to keep the backslash version :sweat_smile:

File: .github/workflows/update-docs.yml (hunk 2 above)
Comment (aaperis) [reply .github/workflows/update-docs.yml]: Looks more reasonable :-)

**Review by kusalananda — APPROVED** [review]

**Review by pontus — APPROVED** [review]

**Review by blankdots — APPROVED** [review]

**Pull request merged**